### PR TITLE
Issue 5889: Make documentation compatible with Docusaurus

### LINKED ISSUE #5889: Make documentation compatible with Docusaurus
**Is your feature request related to a problem? Please describe.**
Docusaurus uses Markdown documentation with minor differences from mkdocs.

**Describe the solution you'd like**
Add Yaml front-matter. Move title from h1 header to front-matter title field. Add sidebars.js.

**Describe alternatives you've considered**
I considered run-time conversion but this does not appear to be worth the effort right now.

**Additional context**

### PR DESCRIPTION
**Pull request by Tristan1900**
**Change log description**  
Make documentation compatible with Docusaurus

**Purpose of the change**  
Fix #5889 

**What the code does**  
Add sidebar.js
Add doc header for every file
Escape special characters in restAPI.md, add new line after html anchor.
Add pravega-write-read-methods.md

**How to verify it**

### CODE REVIEW

**Review by claudiofahey — APPROVED** [review]

File: documentation/src/docs/pravega-write-read-methods.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+title: Methods for Writing and Reading Pravega Streams
+---
+
+import { IfHaveFeature, IfMissingFeature, SDPVersion } from 'nautilus-docs';
+
+## Overview
+
+This table describes the methods that can be used to write and read data in Pravega streams. It is intended to be a comprehensive list which includes open-source, experimental, and commercial products that integrate with Pravega.
+
+Method                                                                                | Write | Transactional Write | Read | Special Features | Notes
```
Comment (derekm) [inline documentation/src/docs/pravega-write-read-methods.md:11]: We could add:

**Debezium**
https://debezium.io/documentation/reference/operations/debezium-server.html#_pravega
Write: Yes
Transactional Write: Yes
Read: No

File: documentation/src/docs/pravega-write-read-methods.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+[Kafka Adapter](https://github.com/pravega/kafka-adapter)                             | Yes   | No                  | No   |                  | Limited support for Kafka applications that use Java.
+[Boomi](https://github.com/pravega/boomi-connector)                                   | Yes   | No                  | Yes  |                  |
+[Pravega GRPC Gateway](https://github.com/pravega/pravega-grpc-gateway)               | Yes   | Yes                 | Yes  |                  |
+[GStreamer](https://github.com/pravega/gstreamer-pravega)                             | Yes   | No                  | Yes  |                  |
+[NVIDIA DeepStream Message Broker](https://github.com/pravega/gstreamer-pravega)      | Yes   | No                  | No   |                  |
+[.NET](https://github.com/rofr/pravega-sharp)                                         | Yes   | No                  | Yes  |                  |
+[Akka](https://doc.akka.io/docs/alpakka/current/pravega.html)                         | Yes   | No                  | Yes  |                  |
```
Comment (derekm) [inline documentation/src/docs/pravega-write-read-methods.md:28]: Special Features: KVT

File: documentation/src/docs/pravega-write-read-methods.md (hunk 1 above)
Comment (derekm) [reply documentation/src/docs/pravega-write-read-methods.md:11]: *ZIO*
https://github.com/cheleb/zio-pravega
Write: Yes
Transactional Write: No
Read: Yes

File: documentation/src/docs/pravega-write-read-methods.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Method                                                                                | Write | Transactional Write | Read | Special Features | Notes
+--------------------------------------------------------------------------------------|-------|---------------------|------|------------------|------------------------------------------------------
+[Pravega Sensor Collector](#pravega-sensor-collector)                                 | Yes   | Yes                 | No   | EO               |
+[SDP Ingest Gateway](#sdp-ingest-gateway)                                             | Yes   | No                  | No   |                  |
+[Pravega Ingest Gateway](https://github.com/pravega/pravega-ingest-gateway)           | Yes   | No                  | Yes  |                  |
+[Pravega Search](#pravega-search)                                                     | Yes   | Yes                 | Yes  | EO               |
+[Apache Flink](../flink-connectors/overview.md)                                       | Yes   | Yes                 | Yes  | EO               |
```
Comment (derekm) [inline documentation/src/docs/pravega-write-read-methods.md:17]: We need to bring back submodule docs so these "../{flink,spark}-connectors/" links work...

File: documentation/src/docs/pravega-write-read-methods.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+## SDP Ingest Gateway
+
+SDP Ingest Gateway is a component of **Dell EMC Streaming Data Platform** that provides a REST interface for writing events into Pravega streams. It is designed to receive POST requests from a variety of sources including Dell iDRAC, a management controller card embedded in the motherboard of Dell PowerEdge servers.
+
+For more information, see the [Streaming Data Platform Developer's Guide](https://dl.dell.com/content/docu96951_Streaming_Data_Platform_Developers_Guide.pdf).
+
+## Pravega Search
```
Comment (derekm) [inline documentation/src/docs/pravega-write-read-methods.md:54]: This whole section and its listing can be disabled on the open-source side.

File: documentation/src/docs/pravega-write-read-methods.md (hunk 3 above)
Comment (derekm) [reply documentation/src/docs/pravega-write-read-methods.md:17]: Can we get these docs to render from here: https://github.com/pravega/docs such that these links work?

**Review by derekm — CHANGES_REQUESTED** [review]
Can we include rendering the open-source docs from: https://github.com/pravega/docs as within the scope of this PR?

Can we look into porting the mkdocs theme to Docusaurus?
https://github.com/pravega/pravega/tree/master/documentation/src/custom_theme
https://pravega.io/docs/nightly/

We need to replace `./gradlew mkdocsBuild` in build.gradle with a Docusaurus equivalent.

**Review by derekm — APPROVED** [review]
I just noticed this is requesting merge into a `docusaurus` branch in the Pravega repo.

With Wenqi leaving us, I guess we can accept this more or less as-is, and reopen the PR under the new branch.

I'm looking at removing mkdocs from the build and introducing new docusaurus commands using gradle npm plugins.

**Pull request merged**